### PR TITLE
More block: Match the design in the editor with the front, resolve the undo/redo issue

### LINKED ISSUE #34419: More-links lack consistency in input (backend) and output (frontend)
To insert a more-link in the block editor:
1. type `/more` in an article
2. a dashed line with capitalized **"READ MORE"** appears

In the frontend the text **"(more...)"** is printed by the function `the_content();` (without any arguments).

In the frontend the text **"some text"** is printed by the function `the_content(‘some text’);`.

## The problem
The read more-text prints inconsistently in backend ("READ MORE") and frontend ("(more...)" or something else), while the backend is supposed to mimic the frontend.

### Additional problems
- To content creators it is unclear that you can actually customize the read more-text
- The capitalization suggests that this text should be capitals, and will be printed in the frontend as capital, which is not the case.
- Content and presentation are not separated properly

## Proposed solutions
1. If nothing else would change, then at least the placeholder text should be “(more..)”, not capitalized, by default
2. A better solution would be that the argument in `the_content(‘argument’);` would be printed in the backend
3. The default read more text could appear as a an option in the theme customizer

### PR DESCRIPTION
**Pull request by carolinan**
## What?

With this pull request, my intention was first to only update the **look** of the block in the editor to match the front. But it turned into a small refactor.

When I tested the block, I found that changing it from using an `input` element to use a `PlainText` (v2) would also solve the problem with the undo/redo, described in issue 58875.

Closes  https://github.com/WordPress/gutenberg/issues/34419, https://github.com/WordPress/gutenberg/issues/58875

## Why?

The block was styled to look like the more tag in the classic editor. This made the block recognisable for user who has used the classic editor. But the block style did not match the front, which arguably made it more difficult to see what the end result would be.

## How?

- Updates the default text in the editor to match the default text on the front. The new text is` (more...)`.
- Removes the editor styles, since the styles made the block look different than on the front.
- Replaces the `input` element with a `PlainText` component with an `a` tag, because the `input` element has some default styles that would need to be overwritten.
- Replaces the `onKeyDown` behaviour with `disableLineBreaks`, and uses `__unstablOnSplitAtEnd` to insert a new paragraph block if the user clicks on `Enter` at the very end of the block's text content.

## Testing Instructions

Activate a block theme.

Deactivate Gutenberg.
Create a new post.
Add some content
Add a more block. Adjust the text and then try to use the undo commands.
Add some content after the more block.

Activate Gutenberg with this PR.
Confirm that there are no block validation errors caused by the changes to the more block.
Confirm that the more block has a new design, it should now look like a plain text with a link.
Adjust the text and then try to use the undo commands. Confirm that the undo now works.

Next, test the block in an archive, using a query loop with a post content block.
It must be a post content block, it will not work with an excerpt.
View the archive on the front, locate the post you created, and confirm that:
- The content added before the more block shows
- The more link shows.
- The content after the more block is not shown.
- Clicking the more link brings you to the post, and the browser scrolls to the content **after** the more block. The more block itself should not be visible on the single post view.

## Screenshots or screencast 
Before:
More block in the editor, centered with capitalized letters, dashes on the left and right, and with a white background.

After:
More block in the editor, which looks like a plain linked text

### CODE REVIEW

File: packages/block-library/src/more/edit.js [hunk 1]
```diff
@@ -54,14 +40,22 @@ export default function MoreEdit( {
...
+					aria-label={ __( '“more” link text' ) }
+					value={ customText || DEFAULT_TEXT }
 					placeholder={ DEFAULT_TEXT }
-					onChange={ onChangeInput }
-					onKeyDown={ onKeyDown }
-					style={ style }
+					onChange={ ( value ) =>
+						setAttributes( { customText: value } )
+					}
+					allowedFormats={ [] }
```
Comment (ellatrix) [inline packages/block-library/src/more/edit.js]: If we are disabling all formats, it's better to use PlainText v2 (we should find something better than the v2 at some point)

**Review by ellatrix — APPROVED** [review]
Great, we should almost never use input fields in block markup, unless it's an actual input on the front end.

File: packages/block-library/src/more/edit.js (hunk 1 above)
Comment (carolinan) [reply packages/block-library/src/more/edit.js]: Does it support disableLineBreaks and __unstableOnSplitAtEnd? 🤔

Comment (ellatrix) [reply packages/block-library/src/more/edit.js]: Yes :)

Comment (carolinan) [reply packages/block-library/src/more/edit.js]: OK, I was looking at the Plaintext docs, not EditableText docs  👍

**Comment by carolinan** [comment]
I will update the tests later this week because of a national holiday.

**Review by richtabor — CHANGES_REQUESTED** [review]
I don't think we should reduce the utility design of the more block. The other refactoring is good, but as a utility block it should clearly communicate the intention (essentially a page break). 

I left a comment on the main issue: https://github.com/WordPress/gutenberg/issues/34419#issuecomment-2085277130

**Comment by richtabor** [comment]
I think it'd be helpful to maintain the page break design, but use the colors from the site, so that it ensures proper contrast (rather than the arbitrary #ccc and white).

**Pull request closed without merging**